### DISCUSSION TITLE
Time running ReportGenerator on large dir of files

### DISCUSSION THREAD

**Discussion by willcooley4** [danielpalme/ReportGenerator#555] [Q&A] [answered]
I've recently been attempting to scale up use of ReportGenerator's ability to use globbing to generate a combined html report on a large (~800 .coveragexml) directory, and it has been taking a lot of time to run, at least 16h before running into the time limit of the box its on. 
I was wondering if this is expected behavior at this scale? Would filtering out certain files from coverage significantly help this runtime?

**Comment by danielpalme** [comment, marked as answer]
In generell I recommend:

- Try to produce as few coverage files as possible. E.g. instead of running the test for each assembly/module separately, try to run them in batches or in one execution. Reducing the number of files will reduce merging effort of _ReportGenerator_
- If fewer coverage files are not an option for your scenario, you could generate separate coverage reports for different parts of your application(s). Each report should only need a subset of the total 800 files.
- If your machine has enough memory, you can experiment with the following [settings](https://github.com/danielpalme/ReportGenerator/pull/262):
  - settings:numberOfReportsParsedInParallel=1
  - settings:numberOfReportsMergedInParallel=1
  
  Maybe the following is a good start: `settings:numberOfReportsParsedInParallel=10 settings:numberOfReportsMergedInParallel=2`

See also:
https://github.com/danielpalme/ReportGenerator/pull/262

**Reply by willcooley4** [reply]
Thanks! Running tests in batches brought it down to about 1h30 :)

**Reply by danielpalme** [reply]
Great!
